docs/guides/software-lifecycle/security/dependency-vulnerability-scanning/README.mdx

```diff
@@ -0,0 +1,127 @@
+
+import CodeBlock from '@theme/CodeBlock';
+import PreCommitConfigSource from '!!raw-loader!./.pre-commit-config.yml';
+
+# Dependency Vulnerability Scanning
+
+<pre align="center">A guide to scanning packaged software dependencies and container images for security vulnerabilities both manually and automatically.</pre>
+
+![banner-image](/img/vulnerability-scanning-screen.png)
+
+## Introduction
+
+**Background**: To maintain the integrity and security of your applications in production environments, it's essential to monitor dependency vulnerabilities. Third-party software dependencies can harbor security vulnerabilities. This guide focuses on utilizing Grype, an open source vulnerability scanner, to proactively detect vulnerabilities in dependencies defined within packages and container images.
+
+**Use Cases**:
+- Scanning container images for vulnerabilities during the development phase
+- Ensuring base container images are as vulnerability-free as possible
+- Scanning package-manager defined software dependencies (e.g. NPM, YARN, Maven, etc.) for vulnerabilities during the development phase
+- Automating vulnerability detection in repositories
+
+---
+
+## Prerequisites
+**Software:**
+- OCI compliant containers (e.g. Docker, Podman) or other package-manager software dependencies
+- `pre-commit` framework
+
+**Skills:**
+- Basic knowledge of Git hooks and Docker commands
+- Understanding of YAML for pre-commit configuration
+
+---
+
+## Quick Start
+
+**Run a local scan of your container's repository (folder containing the Dockerfile) using [Grype](https://github.com/anchore/grype)**
+
+```bash
+grype dir:.
+```
+
+**⬇️ [.pre-commit-config.yml](.pre-commit-config.yml)**
```

Comment:
If you're going to call YAML files `.yml`, I will insist you call HTML files `.htm` 😈

Reply:
I hear you @nutjob4life! However, this is a "feature" of the pre-commit tool itself. It doesn't recognize `yaml` extensions, it is strictly looking for a `.pre-commit.yml` file in the local repository. There is a workaround, but its extra typing. See https://github.com/pre-commit/pre-commit/issues/702

Reply:
Hi @riverma, unless my dyslexia is kicking in, I think you've got that reversed. The pre-commit tool looks specifically for `.pre-commit-config.yaml` (with the `a`).

The [issue cited](https://github.com/pre-commit/pre-commit/issues/702) wanted support without the `a` (`.yml`) but the maintainers were all "nah bro".

Reply:
@nutjob4life - apologies! I keep making the mistake of citing things that actually contradict my argument 😂. I could’ve sworn I got an error message from pre-commit though. Let me do some digging and share feedback.

BTW what is your opinion on .jpg vs .jpeg? ;)

Reply:
> BTW what is your opinion on .jpg vs .jpeg? ;)

@riverma don't get me started 🤣

But I will leave you with this:

!["GIF"](https://github.com/NASA-AMMOS/slim/assets/814813/b78ecab6-699f-4062-8c33-53a100e3837d)

Reply:
🤯